test/test_energy_storage_device.py

```diff
@@ -411,21 +439,47 @@ def test_impedance(self):
             while line.find('(Rad/s)') == -1:
                 line = fout.readline()
 
-            # extract the output
+            # output is organized in sections of decreasing
+            # Omega values. Organize into chunks
             while line != '':
-                if line.find(',') != -1:
-                    tmp = line.split(',')
-                    # indication of failure: entry is 'NaN'
+                if line.find('.') != -1:
+                    # first section not separated by commas
+                    if line.find(',') == -1:
+                        tmp = line.lstrip(' ').split()
+                    else:
+                        tmp = line.split(',')
                     omega.append(float(tmp[0]))
                     z_real.append(float(tmp[1]))
                     z_imag.append(float(tmp[2]))
+                else:
+                    # must be in between a section
+                    # if we have output, add it as a section
+                    if len(omega) != 0:
+                        omega_sections.append(omega)
+                        z_real_sections.append(z_real)
+                        z_imag_sections.append(z_imag)
+                        omega.clear()
+                        z_real.clear()
+                        z_imag.clear()
                 line = fout.readline()
 
-        # indication of failure: an entry that is not nonzero
-        for i in range(0, len(omega)-1):
-            self.assertNotEqual(omega[i], 0)
-            self.assertNotEqual(z_real[i], 0)
-            self.assertNotEqual(z_imag[i], 0)
+        # Assert the following relationships hold:
+        # (1) `Omega` decreases accross each section
+        # (2) `Z_real` increases accross each section
+        # (3) `z_imag` decreases accross each section
```

Comment:
Again some problem. There is no tolerance on this check. It's just to easy to satisfy it.